src/Microsoft.OpenApi/Services/OpenApiFilterService.cs

```diff
@@ -0,0 +1,161 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.OpenApi.Models;
+
+namespace Microsoft.OpenApi.Services
+{
+    /// <summary>
+    /// A service that slices an OpenApiDocument into a subset document
+    /// </summary>
+    public static class OpenApiFilterService
+    {
+        /// <summary>
+        /// Create predicate function based on passed query parameters
+        /// </summary>
+        /// <param name="operationIds">Comma delimited list of operationIds or * for all operations.</param>
+        /// <returns>A predicate.</returns>
+        public static Func<OpenApiOperation, bool> CreatePredicate(string operationIds)
+        {
+            string predicateSource = null;
+
+            Func<OpenApiOperation, bool> predicate;
+            if (operationIds != null)
+            {
+                if (operationIds == "*")
+                {
+                    predicate = (o) => true;  // All operations
+                }
+                else
+                {
+                    var operationIdsArray = operationIds.Split(',');
+                    predicate = (o) => operationIdsArray.Contains(o.OperationId);
+                }
+
+                predicateSource = $"operationIds: {operationIds}";
+            }
+
+            else
+            {
+                throw new InvalidOperationException("OperationId needs to be specified.");
+            }
+
+            return predicate;
+        }
+
+        /// <summary>
+        /// Create partial OpenAPI document based on the provided predicate.
+        /// </summary>
+        /// <param name="source">The target <see cref="OpenApiDocument"/>.</param>
+        /// <param name="predicate">A predicate function.</param>
+        /// <returns>A partial OpenAPI document.</returns>
+        public static OpenApiDocument CreateFilteredDocument(OpenApiDocument source, Func<OpenApiOperation, bool> predicate)
+        {
+            // Fetch and copy title, graphVersion and server info from OpenApiDoc
+            var subset = new OpenApiDocument
+            {
+                Info = new OpenApiInfo()
+                {
+                    Title = source.Info.Title + " - subset",
+                    Version = source.Info.Version
+                },
```

Comment:
All the Info properties should be copied over 
![image](https://user-images.githubusercontent.com/447694/139772728-aea0f2df-4a6d-4149-8bda-89b6dcbb3330.png)

Reply:
`Extensions = source.Info.Extensions` ??

Reply:
Yes, good catch.